tests/contrib/celery/test_chained_task.py

```diff
@@ -0,0 +1,46 @@
+import os
+import re
+import signal
+import subprocess
+import time
+
+
+# Ensure that when we call Celery chains, the root span has celery specific span tags
+# The test_integration.py setup doesn't perfectly mimic the condition of a worker process running.
+# This test runs the worker as a side so we can check the tracer logs afterwards to ensure expected span results.
+# See https://github.com/DataDog/dd-trace-py/issues/11479
+def test_task_chain_task_call_task():
+    celery_worker_cmd = "ddtrace-run celery -A tests.contrib.celery.tasks worker -c 1 -l DEBUG -n uniquename1 -P solo"
+    celery_task_runner_cmd = "ddtrace-run python3.11 tests/contrib/celery/run_tasks.py"
+    worker_process = subprocess.Popen(
+        celery_worker_cmd.split(),
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        preexec_fn=os.setsid,
+        close_fds=True,
+        cwd=str(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__))))),
+    )
+
+    # Wait a few seconds for the celery worker to start up before trying to run tasks
+    time.sleep(5)
```

Comment:
oof, can we use something like `celery.task.control.Inspect.stats()` instead?  Sleep-for-external-service are a great source of flakiness.

```
while celery.task.control.Inspect.stats() != worker and max_time not waited:
    sleep_a_small_amount
```

Reply:
Thanks! I'll look into that. Maybe that's why it's failing in CI sometimes but not locally!

Reply:
Resolving since this is fixed in the recent commit.